internal/command/providers_lock.go

```diff
@@ -243,9 +245,17 @@ func (c *ProvidersLockCommand) Run(args []string) int {
 		}
 		ctx := evts.OnContext(ctx)
 
+		// We can not use c.providerGlobalCacheDir() as we install the provider into a temp dir
 		dir := providercache.NewDirWithPlatform(tempDir, platform)
 		installer := providercache.NewInstaller(dir, source)
 
+		// Use global plugin cache for extra speed if this architecture matches the systems (and therefore the caches) one
+		globalCacheDir := c.providerGlobalCacheDir()
```

Comment:
should we add a warning if they've asked for the pluginCache and the platforms don't match?

Reply:
The part about matching the system architecture is outdate since we now do `globalCacheDir.WithPlatform(platform)`, I'll remove it